core/src/main/java/com/linecorp/armeria/common/AbstractHttpRequestBuilder.java

```
@@ -411,42 +425,142 @@ private RequestHeaders requestHeaders() {
 
     private String buildPath() {
         checkState(path != null, "path must be set.");
```

Could we early return path itself if `pathParams.isEmpty()` and `queryParams == null`?

---

I think we still should raise an exception when a user specified `{param}` in a path and `param` doesn't exist.

---

That makes sense.